netflix/src/components/top/TopBar.tsx

```diff
@@ -1,32 +1,40 @@
-import React, { useCallback, useState } from "react";
+import React from "react";
 import Button from "../util/Button";
-import { ButtonType } from "../../interfaces/Button.interface";
-import AddMovie from "../filmInteraction/add/AddMovie";
+import { ButtonType } from "../../interfaces/components/util/Button.interface";
+import { useDispatch } from "react-redux";
+import { renderAdd } from "../../redux/forms";
+import { search } from "../../redux/api";
+import { setFilms } from "../../redux/films";
 
 const TopBar = () => {
-  const [addMovie, setAddMovie] = useState(false);
-  const handleFormState = useCallback(() => {
-    setAddMovie(!addMovie);
-  }, [addMovie]);
+  const dispatch = useDispatch();
+  const [trigger] = search();
+
+  const handleSearch = async (event: any) => {
+    event.preventDefault();
+    const searchString = event.target[0].value;
+    const searchResults = await trigger(searchString).unwrap();
```

Comment:
TEchnically same issue here as above:
 - your selected `genre` and `sortBy` is not taken into consideration... you either need to reset these filters or send them with request